base/linalg/rowvector.jl

```diff
@@ -0,0 +1,183 @@
+immutable RowVector{T,V<:AbstractVector} <: AbstractMatrix{T}
```

Comment:
Of course, that necessitates a bit of care, since the assumption `AbstractVector == ColumnVector` appears in a number of methods, but it doesn't seem too bad.  There would need to be specialized `+`, `*`, and (most annoying) various `broadcast_foo` methods.